### PR TITLE
Add toString() with type for each NavigationAction

### PR DESCRIPTION
**Pull request by jamesreggio**
There's a [growing](https://github.com/redux-saga/redux-saga/issues/732) [trend](https://github.com/reactjs/redux/issues/628#issuecomment-135077511) of defining a custom `toString` to return the type of a given action creator function. This is a convenience that alleviates the need to import both an action creator and type constant.

This simple PR adds a custom `toString` to the action creator factory, and adds basic tests to avoid regressing this behavior in the future.

### CODE REVIEW

File: src/NavigationActions.js [hunk 1]
```diff
@@ -11,10 +11,15 @@ const RESET = 'Navigation/RESET';
...
-  type,
-  ...payload,
-});
+const createAction = (type: string) => {
+  const fn = (payload: Object = {}) => ({
+    type,
+    ...payload,
+  });
+
+  (fn: any).toString = () => type;
```
Comment (matthamil) [inline src/NavigationActions.js]: Nit: Change the Flow type of `fn` to `mixed`. See https://flow.org/en/docs/types/functions/#toc-function-type

Comment (jamesreggio) [inline src/NavigationActions.js]: The type-checker throws the following error.

```
src/NavigationActions.js:20
 20:   (fn: mixed).toString = () => type;
                   ^^^^^^^^ property `toString`. Property cannot be assigned on
 20:   (fn: mixed).toString = () => type;
            ^^^^^ mixed
```

To my understanding, the `mixed` type is used to represent a function with arbitrary argument and return types. The argument/return types of the function are a separate concern from assignment to a property on the function.

I think the only valid typecast here would be to a `Stringifyable` interface that defines a `toString` field. However, I don't believe something like that exists out-of-the-box, and it seems like overkill to add it for this use case.

**Comment by jamesreggio** [comment]
@matthamil, I think this is ready to go. I justified why your proposed changes are not tenable in https://github.com/react-community/react-navigation/pull/2260#discussion_r130895039.

**Comment by jamesreggio** [comment]
Ping @matthamil. This is a trivial change, and I've addressed your concerns.

**Comment by vonovak** [comment]
@jamesreggio we had some changes that now caused conflicts. Could you please resolve them so we can merge this? Thanks!

**Comment by jamesreggio** [comment]
Hi @vonovak, I fixed the merge conflicts. Should be all ready to go.

**Comment by vonovak** [comment]
@jamesreggio does it make sense to document this?

**Comment by jamesreggio** [comment]
@vonovak — I think it's possible that the docs will confuse more than help. Returning the action type from `toString()` is more of a convenience that causes these methods to 'just do the right thing' when used idiomatically with other libraries. However, I took my best stab at describing this in the docs.

If you think that they help, this is ready to merge. If you think they're more harmful than helpful, I'm happy to remove that commit from the branch.

File: src/NavigationActions.js (hunk 1 above)
Comment (vonovak) [reply src/NavigationActions.js]: I think the given explanation justifies the usage of `any`, so I'll consider this resolved.

**Comment by vonovak** [comment]
thanks for the contribution @jamesreggio !

**Pull request merged**